In [1]:
# importacion general de librerias y de visualizacion (matplotlib y seaborn)
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

plt.style.use('default') # haciendo los graficos un poco mas bonitos en matplotlib
#plt.rcParams['figure.figsize'] = (20, 10)

sns.set(style="whitegrid") # seteando tipo de grid en seaborn

In [6]:
types= {
    "event": "category", 
    "model": "category", 
    "condition": "category", 
    "color": "category", 
    "storage": "category"
}
df_events = pd.read_csv('./fiuba-trocafone-tp1-final-set/events.csv', dtype=types, low_memory=False)

In [40]:
df_events.columns

Index(['timestamp', 'event', 'person', 'url', 'sku', 'model', 'condition',
       'storage', 'color', 'skus', 'search_term', 'staticpage',
       'campaign_source', 'search_engine', 'channel', 'new_vs_returning',
       'city', 'region', 'country', 'device_type', 'screen_resolution',
       'operating_system_version', 'browser_version'],
      dtype='object')

In [41]:
df_events.head()

timestamp            event    person                        url  \
0  2018-05-31 23:38:05  ad campaign hit  0004b0a2  /comprar/iphone/iphone-5s   
1  2018-05-31 23:38:05     visited site  0004b0a2                        NaN   
2  2018-05-31 23:38:09   viewed product  0004b0a2                        NaN   
3  2018-05-31 23:38:40         checkout  0004b0a2                        NaN   
4  2018-05-29 13:29:25   viewed product  0006a21a                        NaN   

     sku              model condition storage           color skus  \
0    NaN                NaN       NaN     NaN             NaN  NaN   
1    NaN                NaN       NaN     NaN             NaN  NaN   
2   2694          iPhone 5s       Bom    32GB  Cinza espacial  NaN   
3   2694          iPhone 5s       Bom    32GB  Cinza espacial  NaN   
4  15338  Samsung Galaxy S8       Bom    64GB         Dourado  NaN   

         ...        search_engine channel new_vs_returning        city  \
0        ...                  NaN     NaN              NaN         NaN   
1        ...                  NaN    Paid              New  Camaragibe   
2        ...                  NaN     NaN              NaN         NaN   
3        ...                  NaN     NaN              NaN         NaN   
4        ...                  NaN     NaN              NaN         NaN   

       region country device_type screen_resolution operating_system_version  \
0         NaN     NaN         NaN               NaN                      NaN   
1  Pernambuco  Brazil  Smartphone           360x640                Android 6   
2         NaN     NaN         NaN               NaN                      NaN   
3         NaN     NaN         NaN               NaN                      NaN   
4         NaN     NaN         NaN               NaN                      NaN   

    browser_version  
0               NaN  
1  Chrome Mobile 39  
2               NaN  
3               NaN  
4               NaN  

[5 rows x 23 columns]

In [7]:
df_events.dtypes

timestamp                     object
event                       category
person                        object
url                           object
sku                           object
model                       category
condition                   category
storage                     category
color                       category
skus                          object
search_term                   object
staticpage                    object
campaign_source               object
search_engine                 object
channel                       object
new_vs_returning              object
city                          object
region                        object
country                       object
device_type                   object
screen_resolution             object
operating_system_version      object
browser_version               object
dtype: object

In [30]:
df_events['event'].describe()

count            1011288
unique                11
top       viewed product
freq              528931
Name: event, dtype: object

#### Como hay solo 11 valores distintos lo tomo como un categorico

In [28]:
df_events['model'].describe()

count       564284
unique         202
top       iPhone 6
freq         50916
Name: model, dtype: object

#### Como hay solo 202 valores distintos, pero se contaron 564284 valores, lo tomo como un categorico

In [32]:
df_events.describe()

timestamp           event    person    url       sku  \
count               1011288         1011288   1011288  82756  563838.0   
unique               793805              11     27624    227    3885.0   
top     2018-05-15 15:56:06  viewed product  71492f2b      /    2830.0   
freq                     11          528931      2771  28323    4311.0   

           model condition storage   color  \
count     564284    563836  563836  563836   
unique       202         5       8      63   
top     iPhone 6       Bom    16GB   Preto   
freq       50916    243014  190833  132960   

                                                     skus       ...        \
count                                              221699       ...         
unique                                              35310       ...         
top     2820,6706,6720,2750,6649,7251,6663,12604,7224,...       ...         
freq                                                  594       ...         

       search_engine channel new_vs_returning     city     region country  \
count          50957   87378            87378    87378      87378   87378   
unique             4       7                2     1939         93      46   
top           Google    Paid        Returning  Unknown  Sao Paulo  Brazil   
freq           50240   44193            60480    15819      24996   84308   

       device_type screen_resolution operating_system_version browser_version  
count        87378             87378                    87378           87378  
unique           4               282                      121             343  
top     Smartphone           360x640               Windows 7      Chrome 66.0  
freq         44239             30009                    19675           22611  

[4 rows x 23 columns]